pyodide-build/pyodide_build/common.py

```diff
@@ -348,18 +350,68 @@ def search_pyodide_root(curdir: str | Path, *, max_depth: int = 5) -> Path:
     )
 
 
-def init_environment() -> None:
+def init_environment(*, quiet: bool = False) -> None:
+    """
+    Initialize Pyodide build environment.
+    This function needs to be called before any other Pyodide build functions.
+    """
     if os.environ.get("__LOADED_PYODIDE_ENV"):
         return
+
     os.environ["__LOADED_PYODIDE_ENV"] = "1"
+
+    _set_pyodide_root(quiet=quiet)
+
+
+def _set_pyodide_root(*, quiet: bool = False) -> None:
+    """
+    Set PYODIDE_ROOT environment variable.
+
+    This function works both in-tree and out-of-tree builds:
+    - In-tree builds: Searches for the root of the Pyodide repository in parent directories
+    - Out-of-tree builds: Downloads and installs the Pyodide build environment into the current directory
+
+    Parameters
+    ----------
+    quiet
+        If True, do not print any messages
+    """
+
+    from . import install_xbuildenv  # avoid circular import
+
     # If we are building docs, we don't need to know the PYODIDE_ROOT
     if "sphinx" in sys.modules:
         os.environ["PYODIDE_ROOT"] = ""
+        return
 
+    # 1) If PYODIDE_ROOT is already set, do nothing
     if "PYODIDE_ROOT" in os.environ:
-        os.environ["PYODIDE_ROOT"] = str(Path(os.environ["PYODIDE_ROOT"]).resolve())
-    else:
-        os.environ["PYODIDE_ROOT"] = str(search_pyodide_root(os.getcwd()))
+        return
```

Comment:
I think this guard should be the very first thing in the function body. If `_set_pyodide_root` is only called from `init_environment` this might be redundant to the guard on `init_environment`. Though perhaps it is still good for readability.

Reply:
I think the `__LOADED_PYODIDE_ENV` guard in `init_environment` is a bit redundant, but I didn't want to introduce an unexpected side effect. I'll check it in a subsequent PR.